jscomp/ml/ctype.ml

```diff
@@ -3939,8 +3939,14 @@ let rec subtype_rec env trace t1 t2 cstrs =
               let (co, cn) = Variance.get_upper v in
               if co then
                 if cn then
-                  (trace, newty2 t1.level (Ttuple[t1]),
-                   newty2 t2.level (Ttuple[t2]), !univar_pairs) :: cstrs
+                  (* Invariant type argument: check both ways *)
+                  if
+                    subtype_rec env ((t1, t2)::trace) t1 t2 [] = [] &&
```

Comment:
This is using the special case `cstrs = []`. Not sure if there's a good way to use existing constraints collected up to now.